File: easy/two_sum/two_sum.py
from typing import List

def two_sum(nums: List[int], target: int) -> List[int]:
    d = {}
    ret = []
    for i, j in enumerate(nums):
        d[i] = j
    for i in nums:
        if target - i in nums:
            try:
                ret.append(nums.index(target - i, nums.index(i) + 1))
                ret.append(nums.index(i))
            except ValueError:
                continue
            break
    return ret

def main():
    i = [int(i) for i in input().split()]
    nums = i[0: -1]
    target = i[-1]
    print(two_sum(nums, target))

if __name__ == '__main__':
    main()

File: easy/two_sum/two_sum.cpp
#include <iostream>
#include <algorithm>
#include <vector>

std::vector<int> two_sum(std::vector<int> &nums, int target) {
   std::vector<int> ret;
   std::vector<int> sorted_nums = nums;
   std::sort(sorted_nums.begin(), sorted_nums.end());
   for (auto i = sorted_nums.begin(), j = sorted_nums.end() - 1; i != j; ) {
      if (*i + *j == target) {
         auto i_it = std::find(nums.begin(), nums.end(), *i);
         auto j_it = std::find(nums.begin(), nums.end(), *j);
         if (*i == *j) {
            j_it = std::find(i_it + 1, nums.end(), *j);
         }
         ret.push_back(i_it - nums.begin());
         ret.push_back(j_it - nums.begin());
         return ret;
      } else if (*i + *j < target) {
         i++;
      } else {
         j--;
      }
   }
   return ret;
}

int main() {
   std::vector<int> nums;
   int size, target;
   scanf("%d", &size);
   for (int i = 0; i < size; i++) {
      int in;
      scanf("%d", &in);
      nums.push_back(in);
   }
   scanf("%d", &target);
   for (int i: two_sum(nums, target)) {
      std::cout << i << " ";
   }
   std::cout << "\n";
}
